src/signtool/IBM_Container.cpp

```diff
@@ -0,0 +1,1712 @@
+              << std::endl
+              << "   m_hashAlg          = " << GetHashAlgoName( (int) m_hashAlg )
+              << std::endl
+              << "   m_sigAlg           = " << GetSignAlgoName( (int) m_sigAlg )
+              << std::endl;
+
+    std::cout << "   m_codeStartOffset  = " << std::hex << std::setfill('0') << std::setw(16) << m_codeStartOffset
+              << std::endl;
+
+    std::cout << "   m_reserved         = ";
+    printHexBytes( m_reserved, 8, g_COL_WIDTH );
+
+    std::cout << "   m_flags            = " << std::hex << std::setfill('0') << std::setw(8) << m_flags
+              << std::endl
+              << "   m_swKeyCount       = " << (int) m_swKeyCount
+              << std::endl;
+
+    std::cout << "   m_payloadSize      = " << std::hex << std::setfill('0') << std::setw(16) << m_payloadSize
+              << std::endl;
+
+    std::cout << "   m_payloadHash      = ";
+    printHexBytes( m_payloadHash, SHA512_DIGEST_SIZE, g_COL_WIDTH );
+
+    std::cout << "   m_ecidCount        = " <<  (int) m_ecidCount
+              << std::endl;
+
+    for (int j = 0; j < m_ecidCount; j++)
+    {
+        std::cout << "   m_ecid[" << j << "]         = ";
+        printHexBytes( (const uint8_t *) &m_ecid[(j*ECID_SIZE)], ECID_SIZE, 24 );
+    }
+    std::cout << std::endl;
+}
+
+
+/**
+ * @brief   Get the Prefix Data Header message as a byte stream
+ *
+ * @param[out]  packet  -  A reference to a vector of type uint8_t
+ */
+void PrefixData::GetHeaderBytes( std::vector<uint8_t>& packet ) const
+{
+    packet.clear();
+
+    // insert the HW Signature-A field
+    packet.insert( packet.end(),
+                   m_hwSigA,
+                   (m_hwSigA + sizeof(m_hwSigA)) );
+    
+    // insert the HW Signature-B field
+    packet.insert( packet.end(),
+                   m_hwSigB,
+                   (m_hwSigB + sizeof(m_hwSigB)) );
+    
+    // insert the HW Signature-C field
+    packet.insert( packet.end(),
+                   m_hwSigC,
+                   (m_hwSigC + sizeof(m_hwSigC)) );
+
+    uint8_t zeroKey[ECDSA521_KEY_SIZE];
+    memset( zeroKey, 0, ECDSA521_KEY_SIZE );
+
+    // insert the SW Public Key-P field only if its valid
+    if (memcmp( m_swPkeyP, zeroKey, ECDSA521_KEY_SIZE) != 0)
+    {
+        packet.insert( packet.end(),
+                       m_swPkeyP,
+                       (m_swPkeyP + sizeof(m_swPkeyP)) );
+    }
+    
+    // insert the SW Public Key-Q field only if its valid
+    if (memcmp( m_swPkeyQ, zeroKey, ECDSA521_KEY_SIZE) != 0)
+    {
+        packet.insert( packet.end(),
+                       m_swPkeyQ,
+                       (m_swPkeyQ + sizeof(m_swPkeyQ)) );
+    }
+    
+    // insert the SW Public Key-R field only if its valid
+    if (memcmp( m_swPkeyR, zeroKey, ECDSA521_KEY_SIZE) != 0)
+    {
+        packet.insert( packet.end(),
+                       m_swPkeyR,
+                       (m_swPkeyR + sizeof(m_swPkeyR)) );
+    }
```

Comment:
Correction, that is the data structure, it's just not actually bloody documented anywhere, like, say, in the headers describing the container format.